lib/views/push-pull-view.js

```diff
@@ -2,37 +2,196 @@ import React from 'react';
+    {'animate-rotate': isSyncing},
+  );
+}
+
 export default class PushPullView extends React.Component {
   static propTypes = {
-    pushInProgress: PropTypes.bool,
-    fetchInProgress: PropTypes.bool,
+    currentBranch: BranchPropType.isRequired,
+    currentRemote: RemotePropType.isRequired,
+    isSyncing: PropTypes.bool,
     behindCount: PropTypes.number,
     aheadCount: PropTypes.number,
+    push: PropTypes.func.isRequired,
+    pull: PropTypes.func.isRequired,
+    fetch: PropTypes.func.isRequired,
+    originExists: PropTypes.bool,
   }
 
   static defaultProps = {
-    pushInProgress: false,
-    fetchInProgress: false,
+    isSyncing: false,
     behindCount: 0,
     aheadCount: 0,
   }
 
+
+  componentDidMount() {
+    this.setTooltip();
+  }
+
+  componentDidUpdate() {
+    this.tooltip.dispose();
+    this.setTooltip();
+  }
+
+  setTooltip = () => {
+    this.tooltip = atom.tooltips.add(this.tileNode, {
+      title: `<div style="text-align: left; line-height: 1.2em;">${this.tooltipTarget.dataset.tooltip}</div>`,
+      delay: {show: 200, hide: 100},
+      html: true,
+    });
+  }
+
+  onClickPush = clickEvent => {
+    if (this.props.isSyncing) {
+      return;
+    }
+    this.props.push({
+      force: clickEvent.metaKey || clickEvent.ctrlKey,
+      setUpstream: !this.props.currentRemote.isPresent(),
+    });
+  }
+
+  onClickPull = clickEvent => {
+    if (this.props.isSyncing) {
+      return;
+    }
+    this.props.pull();
+  }
+
+  onClickPushPull = clickEvent => {
+    if (this.props.isSyncing) {
+      return;
+    }
+    if (clickEvent.metaKey || clickEvent.ctrlKey) {
+      this.props.push({
+        force: true,
+      });
+    } else {
+      this.props.pull();
+    }
+  }
+
+  onClickPublish = clickEvent => {
+    if (this.props.isSyncing) {
+      return;
+    }
+    this.props.push({
+      setUpstream: !this.props.currentRemote.isPresent(),
+    });
+  }
+
+  onClickFetch = clickEvent => {
+    if (this.props.isSyncing) {
+      return;
+    }
+    this.props.fetch();
+  }
+
   render() {
-    const pushing = this.props.pushInProgress;
-    const pulling = this.props.fetchInProgress;
-    const pushClasses = cx('github-PushPull-icon', 'icon', {'icon-arrow-up': !pushing, 'icon-sync': pushing});
-    const pullClasses = cx('github-PushPull-icon', 'icon', {'icon-arrow-down': !pulling, 'icon-sync': pulling});
-    return (
-      <div className="github-PushPull inline-block" ref={e => { this.element = e; }}>
-        <span className={pullClasses} />
-        <span className="github-PushPull-label is-pull">
-          {this.props.behindCount ? `${this.props.behindCount}` : ''}
+    const isAhead = this.props.aheadCount > 0;
+    const isBehind = this.props.behindCount > 0;
+    const isUnpublished = !this.props.currentRemote.isPresent();
+    const isDetached = !!this.props.currentBranch.detached;
+    const isSyncing = this.props.isSyncing;
+    const hasOrigin = !!this.props.originExists;
+
+    return (<div
```

Comment:
🚨 :art: We prefer this style for multi-line jsx, with parenthesis on their own lines:

```js
return (
  <div
     ref={...}
     className={...}>
  </div>
)
```